### PR TITLE
Update metrics server to 0.4.1

### PR DESCRIPTION
**Pull request by omegas27**
I still need to test it but I could use a pair of extra eyes here

### CODE REVIEW

File: helm/metrics-server-app/templates/api-service.yaml [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
   service:
     name: {{ .Values.name }}
     namespace: {{ .Values.namespace }}
-  group: metrics.k8s.io
   version: v1beta1
-  insecureSkipTLSVerify: {{ .Values.apiService.insecureSkipTLSVerify }}
-  groupPriorityMinimum: 100
   versionPriority: 100
-{{- end -}}
+{{- end -}}
```
Comment (MarcelMue) [inline helm/metrics-server-app/templates/api-service.yaml:17]: Newline is missing.

File: helm/metrics-server-app/templates/_helpers.tpl [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+app.giantswarm.io/branch: {{ .Values.project.branch | quote }}
+app.giantswarm.io/commit: {{ .Values.project.commit | quote }}
+app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+helm.sh/chart: {{ include "chart" . | quote }}
+giantswarm.io/service-type: {{ .Values.serviceType }}
+{{- end -}}
+
+{{/*
+Selector labels
```
Comment (MarcelMue) [inline helm/metrics-server-app/templates/_helpers.tpl:32]: We should verify if upgrades still work like this.

Comment (omegas27) [reply helm/metrics-server-app/templates/_helpers.tpl:32]: Sure, this is what I was planning on testing :)

**Comment by omegas27** [comment]
Upgrade tested on geckon:

```
k logs -n kube-system metrics-server-697f87f65f-gh8mh -f
I1210 09:27:57.830449       1 serving.go:325] Generated self-signed cert (/tmp/apiserver.crt, /tmp/apiserver.key)
I1210 09:27:58.881098       1 requestheader_controller.go:169] Starting RequestHeaderAuthRequestController
I1210 09:27:58.881157       1 shared_informer.go:240] Waiting for caches to sync for RequestHeaderAuthRequestController
I1210 09:27:58.881374       1 configmap_cafile_content.go:202] Starting client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file
I1210 09:27:58.881388       1 shared_informer.go:240] Waiting for caches to sync for client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file
I1210 09:27:58.881415       1 configmap_cafile_content.go:202] Starting client-ca::kube-system::extension-apiserver-authentication::client-ca-file
I1210 09:27:58.881445       1 shared_informer.go:240] Waiting for caches to sync for client-ca::kube-system::extension-apiserver-authentication::client-ca-file
I1210 09:27:58.882145       1 dynamic_serving_content.go:130] Starting serving-cert::/tmp/apiserver.crt::/tmp/apiserver.key
I1210 09:27:58.882422       1 secure_serving.go:197] Serving securely on [::]:10443
I1210 09:27:58.882456       1 tlsconfig.go:240] Starting DynamicServingCertificateController
I1210 09:27:58.981843       1 shared_informer.go:247] Caches are synced for RequestHeaderAuthRequestController 
I1210 09:27:58.982335       1 shared_informer.go:247] Caches are synced for client-ca::kube-system::extension-apiserver-authentication::client-ca-file 
I1210 09:27:58.981922       1 shared_informer.go:247] Caches are synced for client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file 
```

**Review by MarcelMue — APPROVED** [review]

**Comment by omegas27** [comment]
Thank you @MarcelMue  :)

**Pull request merged**